controllers/machine_controller.go

```diff
@@ -385,9 +388,13 @@ func (r *MachineReconciler) reconcileDelete(ctx context.Context, cluster *cluste
 			return true, nil
 		})
 		if waitErr != nil {
-			log.Error(deleteNodeErr, "Timed out deleting node, moving on", "node", m.Status.NodeRef.Name)
+			log.Error(deleteNodeErr, "Timed out deleting node", "node", m.Status.NodeRef.Name)
 			conditions.MarkFalse(m, clusterv1.MachineNodeHealthyCondition, clusterv1.DeletionFailedReason, clusterv1.ConditionSeverityWarning, "")
 			r.recorder.Eventf(m, corev1.EventTypeWarning, "FailedDeleteNode", "error deleting Machine's node: %v", deleteNodeErr)
+			if r.WaitForNodeDeletion {
+				return ctrl.Result{}, deleteNodeErr
+			}
+			log.Info("Waiting for node deletion is disabled, moving on", "node", m.Status.NodeRef.Name)
```

Comment:
Do we want to wait indefinitely here? Should we instead make the timeout configurable?

Reply:
From my POV it still makes sense to keep the Machine object until the node is deleted from the cluster. From what I understand the use-case described in #1446 is about breaking the API by accident, which causes machine deletion to get postponed. But at some point the API will either recover, or the cluster will get deleted entirely.
Does it make sense to wait indefinitely, unless the cluster is deleted? (We could even skip node deletion entirely if the cluster is deleted, since it doesn't matter anymore)

Reply:
The issue referenced in the PR asks for a configurable timeout https://github.com/kubernetes-sigs/cluster-api/issues/5516

We can reason on having this timeout so it will allow default (null), wait indefinitely (0), wait some time (>0)
Also +1 to having a breaking circuit to drop wait when the cluster is deleted

Reply:
The reason behind that issue is a problem we've faced with metal3, and the only proper solution for that is waiting indefinitely.

What timeout would waiting indefinitely (if we implement that as re-queuing reconciliation) use? Or do you want to just retry deletion in the current reconciliation forever? Wouldn't that block the reconciliation loop for other objects?

Reply:
@schrej and CAPV for that matter

Reply:
That makes sense, are we ok proceeding with a field on Machines instead instead of a CLI manager flag?

Reply:
A field with the semantic as Fabrizio described sounds fine to me.

We already have `NodeDrainTimeout` so it would be nicely consistent to have another field like `NodeDeletionTimeout`.